src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -186,6 +189,39 @@ def sort_code_mapping_configs(
                     sorted_configs.insert(0, config)
         return sorted_configs
 
+    def get_latest_commit_sha_from_blame(
+        self,
+        integration_installation: ApiClient,
+        line_no: int,
+        filepath: str,
+        repository: Repository,
+        ref: str,
+    ) -> Optional[str]:
+        commit_sha = ""
+        git_blame_list = integration_installation.get_blame_for_file(
+            repository,
+            filepath,
+            ref,
+            line_no,
+        )
+        if git_blame_list:
+            latest_commit = git_blame_list[0]
+            for blame in git_blame_list:
+                if blame["commit"]["committedDate"] > latest_commit["commit"]["committedDate"]:
+                    latest_commit = blame
```

Comment:
entirely optional: you could sort git_blame_list by ["commit"]["committedDate] and take the last item. _Technically_ this might be faster but it likely doesn't matter.

```
git_blame_list.sort(key=lambda blame:blame["commit"]["committedDate])
commit_sha = git_blame_list[-1]["commit"]["oid"]
```